src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -870,12 +870,10 @@ void addMethod() {
                     i += 2; // jump to the end of the LOOP idiom
                     continue;
                 case NEW_ARRAY:
-                    Class<?> rtype = name.function.methodType().returnType();
-                    if (isStaticallyNameable(rtype)) {
```

Comment:
I removed the `NEW_ARRAY` intrinsic altogether at first, but added it back in the latest commit. I didn't add the check since I was not aware of the restriction.

Good to know, will add a check.

Reply:
Also, please, add a test case for such scenario. It should be trivial: just use a class defined by the test (loaded by `AppClassLoader`).